redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,43 @@
+package redisxtest
+
+import (
+	"context"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// NewMockRedisClient sets up a mock redis cluster, client and sender
+// This should be called from TestMain since the miniredis instance created is locked
+func NewMockRedisClient(ctx context.Context, timeout time.Duration) (client redisx.Syncx, teardown func(), err error) {
```

Comment:
if this is only intended to be used by unit tests, we do not really need to return `teardown` and `err`. just pass in`tb testing.TB` as an arg (so it can be used by both tests and benchmarks), then when an error happens we can just `tb.Fatalf("Failed to ...: %v", err)`, and at the end, before returning, we can just call `tb.Cleanup(teardown)`.

Reply:
@konradreiche that is correct. We have been following the pattern of setup in `TestMain` since places where the mock redis cluster is used tend to be places where it made sense to init there. This approach also gives the flexibility to use this func in both tests directly but also in `TestMain`. However, if we feel strongly that supporting `TestMain` doesn't have much value, then I am happy to change this.